### PR TITLE
🌟 New: Adds extensive ESLint config

### PR DESCRIPTION
**Pull request by aendrew**
This adds an extensive config based off [eslint-config-airbnb-base][1] as heavily-commented YAML to [.eslintrc.yml][2]. I tweaked a few rules (denoted by `Æ` in the file) to reflect conventions I found in `starter-kit`.

 _It is quite likely this will require further refinement as time wears on!_

I also tweaked most JS files to reflect the new linting rules, focusing mainly on cosmetic changes and using single-line overrides otherwise.

## Other changes:

+ :heavy_plus_sign: Adds: babel-eslint, eslint-plugin-babel to optional deps.
+ :shirt: Lint: Everything except client/scripts/*.js passes linting now!

[1]: https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base
[2]: https://github.com/ft-interactive/starter-kit/blob/eslint-config/.eslintrc.yml

### CODE REVIEW

**Comment by kavanagh** [comment]
I didn't realised the airbnb one was so detailed. Perhaps we could start by removing stuff we have no opinion about, and let developers do what they feel is right.

File: client/scripts/top.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable */
+// TODO lint this file?
```
Comment (kavanagh) [inline client/scripts/top.js]: No

**Comment by aendrew** [comment]
Have removed `TODO` comments from `client/scripts`.

**Comment by callumlocke** [comment]
> Perhaps we could start by removing stuff we have no opinion about, and let developers do what they feel is right.

Not sure I agree with this thinking... For me, a central point of linting is to just choose _something_, even for things we have no strong opinion about, to reduce the range of possible errors and inconsistencies across our code.

btw what is the value in rewriting airbnb's rules manually? Why not just extend it and override it?

**Comment by aendrew** [comment]
@callumlocke The argument made by @kavanagh is that strict linting rules will discourage their use, or frustrate people who are trying to code to deadline. I can kind of see the benefit to both arguments — on one hand, it would be good for us devs to have a really strict set of linting rules so that the code we produce is consistent and causes as little cognitive dissonance as possible for those instances we're collaborating on code, while on the other hand, having a more relaxed set of linting rules for everyone else might encourage their use.

Granted, these are all optional dependencies so I'm not entirely sure how likely it is people will use them anyway. 

re: creating a monolithic `.eslintrc`, the idea was to reduce the number of dependencies users will need to download, have all the rules in one place so they're easier to turn on and off, and to ultimately tweak the ruleset to best reflect the needs of the team.

**Comment by aendrew** [comment]
## Discussion: rules to possibly disable

- [ ] `no-console`: Users quite frequently use `console` for debugging.
- [ ] `prefer-template`: Seeing concatenated strings feels like nails on a chalkboard to me now, but I can see some users chafing at being told to use backticks for everything.

(Will update as I come across more issues while working on `starter-kit` this week. Please don't merge until we've discussed the ruleset fully!)

**Comment by aendrew** [comment]
N.b, I've pushed a new commit that removes the YAML file and replaces it with a JS file that consumes all of the Airbnb base rules and then sets them all to warnings.

**Comment by kavanagh** [comment]
👍🚀

**Comment by callumlocke** [comment]
I just ran `fixpack` to normalize the package.json, to reduce future diff noise

**Comment by aendrew** [comment]
I've squashed my commits; merge at will. :shipit:

**Pull request merged**